llvm/test/CodeGen/AIE/aie2/ra/waw_reg_renaming_loop.mir

```diff
@@ -0,0 +1,635 @@
+    %121:ed = COPY $d2
+    %120:ep, %121.sub_dim_count:ed = VST_2D_dmw_sts_w %7.sub_256_lo:mxm, killed %120, killed %121
+    PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  bb.2:
+    liveins: $wl0, $wl1, $wh3, $y2, $y3
+    PseudoRET implicit $lr,  implicit $wl0, implicit $wl1, implicit $wh3, implicit $y2, implicit $y3
+...
+
+# Ignore copy instructions for renaming candidates.
+---
+name:            copy_rename
+alignment:       16
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: copy_rename
+  ; CHECK: bb.0.entry:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $d0, $d2, $p0, $p1, $p2, $r0, $d1_3d
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   LoopStart $r0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   successors: %bb.1(0x40000000), %bb.2(0x40000000)
+  ; CHECK-NEXT:   liveins: $d0, $d2, $p0, $p1, $p2, $d1_3d
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   renamable $p3 = COPY $p0
+  ; CHECK-NEXT:   renamable $d3 = COPY $d0
+  ; CHECK-NEXT:   $wl0, dead $p3, dead $dc3 = VLDA_2D_dmw_lda_w killed $p3, $d3
+  ; CHECK-NEXT:   renamable $p3 = COPY $p1
+  ; CHECK-NEXT:   renamable $d3_3d = COPY $d1_3d
+  ; CHECK-NEXT:   $wl2, dead $p3, dead $dc3, dead $dc7 = VLDA_3D_dmw_lda_w killed $p3, $d3_3d
+  ; CHECK-NEXT:   renamable $x4, dead renamable $l4 = VMAX_LT_D8 killed renamable $x0, renamable $x2, implicit $crvaddsign
+  ; CHECK-NEXT:   renamable $p3 = COPY $p2
+  ; CHECK-NEXT:   renamable $d3 = COPY $d2
+  ; CHECK-NEXT:   dead $p3, dead $dc3 = VST_2D_dmw_sts_w killed $wl4, killed $p3, $d3
+  ; CHECK-NEXT:   PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.2:
+  ; CHECK-NEXT:   PseudoRET implicit $lr
+  bb.0.entry:
+  successors: %bb.1
+  liveins: $r0, $p0, $p1, $p2, $d0, $d1_3d, $d2
+  LoopStart $r0
+  bb.1:
+    successors: %bb.1, %bb.2
+    liveins: $p0, $p1, $p2, $d0, $d1_3d, $d2
+    %110:ep = COPY $p0
+    %111:ed = COPY $d0
+    undef %0.sub_256_lo:mxa, %110:ep, %111.sub_dim_count:ed = VLDA_2D_dmw_lda_w %110, killed %111
+    %99:ep = COPY $p1
+    %100:eds = COPY $d1_3d
+    %1:mxa = COPY %0
+    undef %3.sub_256_lo:mxa, %99:ep, %100.sub_dim_count:eds, %100.sub_hi_dim_then_sub_dim_count:eds = VLDA_3D_dmw_lda_w killed %99, killed %100
+    %7:mxm, %8:el = VMAX_LT_D8 %1, %3, implicit $crvaddsign
+    %120:ep = COPY $p2
+    %121:ed = COPY $d2
+    %120:ep, %121.sub_dim_count:ed = VST_2D_dmw_sts_w %7.sub_256_lo:mxm, killed %120, killed %121
+    PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  bb.2:
+    PseudoRET implicit $lr
+...
+
+# Don't rename superregisters
+# (undef %281.sub_256_lo:vec512, undef %282.sub_256_lo:vec512),
+# if parts of the sub-register (%281.sub_256_hi:vec512, %282.sub_256_hi:vec512)
+# are constant. Thus the "constant" subregister ($wh2) can be shared across both
+# super registers ($x2 -> %281 and %282). Otherwise an unnecessary move would be
+# introduced to move the constant in the newly renamed register.
```

Comment:
Nit: I think the comment is slightly out of date as we have a more generic solution now

Reply:
updated the comment